File: LeetCode/LC-33-Search-In-Roatated-Sorted-Array/Solution.java
class Solution {
    public int search(int[] nums, int target) {
        int index = -1;

        int l = 0 , r = nums.length -1 ;

        while(l <= r){
            int mid = ( l + r )/2;

            if(nums[mid] == target ){
                index = mid;
                break;
            }


            if(  nums[l]  <= nums[mid]){
                if (target > nums[mid] | target < nums[l]){
                    l = mid + 1;
                }
                else{
                    r = mid - 1;
                }
            }
            else{
                if(target < nums[mid] | target > nums[r]){
                    r = mid - 1;
                }
                else{
                    l = mid + 1;
                }
            }
        }

        return index;
    }
}

File: LeetCode/LC-33-Search-In-Roatated-Sorted-Array/Solution.cpp
class Solution {
public:
    int bS(vector<int>& arr,int l,int r,int x){
        if(l<=r){
            int mid=l+r;mid/=2;
            
            if(arr[mid]==x)return mid;
            else if(arr[mid]> x)return bS(arr,l,mid-1,x);
            else return bS(arr,mid+1,r,x);
        }
        return -1;
        
    }
    int search(vector<int>& nums, int target) {
        int ind=-1,size=nums.size();
        for(int i=1;i<size;i++){
            if(nums[i]<nums[i-1]){
                ind=i;
                break;
            }
        }
        
        if(ind==-1){
            return bS(nums,0,size-1,target);
        }
        else{
            int found=bS(nums,0,ind-1,target);
            
            if(found==-1){
                found=bS(nums,ind,size-1,target);
                return found;
            }
            else{
                return found;
            }
        }
        
        
    }
};
